crates/biome_js_analyze/src/lint/nursery/use_import_extensions.rs

```diff
@@ -82,18 +107,39 @@ declare_rule! {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserializable, Deserialize, Serialize, Eq, PartialEq)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UseImportExtensionsOptions {
+    /// A map of custom import extension mappings, where key is inspected file extension
+    /// and value is a pair of `import` extension and `component` import extension
+    pub suggested_extensions: FxHashMap<String, SuggestedExtensionMapping>,
+}
+
+#[derive(Debug, Clone, Default, Deserializable, Deserialize, Serialize, Eq, PartialEq)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct SuggestedExtensionMapping {
+    /// Extension that should be used for module imports
+    pub import: String,
+    /// Extension that should be used for component file imports
+    pub component: String,
+}
+
 impl Rule for UseImportExtensions {
     type Query = Ast<AnyJsImportSpecifierLike>;
     type State = UseImportExtensionsState;
     type Signals = Option<Self::State>;
-    type Options = ();
+    type Options = Box<UseImportExtensionsOptions>;
```

Comment:
Why did you use `Box`?

Reply:
Plenty of other rules which had options used `Box` so I thought it was some sort of common optimization. Would you prefer to just have `type Options = UseImportExtensionsOptions;` here?

Reply:
No, no need. I wanted to know why. It seems that the signature changed and missed it :)

Reply:
Previously, all rule options were grouped into an enum. This enum was used for each rule in the configuration. We had something like:

```rs
enum AnyOptions {
  None, // for rules without options
  Rule1Options,
  Rule4Options,
}
struct Config {
  rule1: AnyOptions
  rule2: AnyOptions
  // ...
}
```

Our configuration wasted a lot of memory. As an optimization I used `Box` for every option larger than a `Box`.

However, I optimized even more by getting rid of the enum and using precise options. Something like:

```rs
enum AnyOptions {
  NoOptions(NoOptions),
  Rule1Options(Rule1Options),
  Rule4Options(Rule4Options),
}
struct Config {
  rule1: Rule1Options
  rule2: NoOptions
  // ...
}
```

So we might consider removing the box when the size of the options is close to a `Box`. Using `Box` for large options still makes sense because a rule is often configured without its options.